importer/helpers/helpers.go

```diff
@@ -113,8 +113,9 @@ func (n *UnixfsNode) AddChild(child *UnixfsNode, db *DagBuilderHelper) error {
 	}
 
 	// Pin the child node indirectly
-	if db.mp != nil {
-		db.mp.PinWithMode(childkey, pin.Indirect)
+	err = db.bcb(childkey, false)
```

Comment:
On Wed, May 27, 2015 at 11:48:14AM -0700, Jeromy Johnson wrote:
> > @@ -113,8 +113,9 @@ func (n *UnixfsNode) AddChild(child *UnixfsNode, db *DagBuilderHelper) error {
> >  	}
> >  
> >  	// Pin the child node indirectly
> > -	if db.mp != nil {
> > -		db.mp.PinWithMode(childkey, pin.Indirect)
> > +	err = db.bcb(childkey, false)
> 
> the bool is for whether or not the given node is the root of its
> respective dag

Do we need this?  Couldn't we just handle root-level changes in the
caller?  This is root-of-the-added-DAG, right?  Not my ‘top’ (tip of a
given layout or an unchunked object) from #1291?  For example,
recursively adding a directory tree will have one root (the final
layout/unchunked directory node that's returned from the adder) but
potentitially many tops (one for each file or directory in the added
tree).